src/certificates/certificates-card/CertificatesCard.test.jsx

```diff
@@ -0,0 +1,104 @@
+import { render, waitFor, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Provider, useDispatch, useSelector } from 'react-redux';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { initializeMockApp } from '@edx/frontend-platform';
+
+import initializeStore from '../../store';
+import { getComponentMode } from '../data/selectors';
+import { MODE_STATES } from '../data/constants';
+import { createCourseCertificate } from '../data/thunks';
+import CertificatesCard from './CertificatesCard';
+import messages from './messages';
+
+let store;
+const courseId = 'course-123';
+const certificateId = 1;
+const signatories = [{
+  name: 'John Doe', title: 'CEO', organization: 'Company', signatureImagePath: '',
+}];
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: jest.fn(),
+  useDispatch: jest.fn(() => jest.fn()),
+}));
+
+jest.mock('../data/thunks', () => ({
+  createCourseCertificate: jest.fn(),
+}));
+
+const renderComponent = () => render(
+  <Provider store={store}>
+    <IntlProvider locale="en">
+      <CertificatesCard
+        id={certificateId}
+        signatories={signatories}
+        courseId={courseId}
+      />
+    </IntlProvider>
+  </Provider>,
+);
+
+const initialState = {
+  certificates: {
+    certificatesData: {
+      certificates: [],
+      hasCertificateModes: true,
+    },
+    componentMode: MODE_STATES.create,
+  },
+};
+
+describe('CertificatesCard', () => {
+  beforeEach(() => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+    store = initializeStore(initialState);
+  });
+
+  it('renders form elements in create mode', () => {
+    useSelector.mockImplementation(selector => {
+      if (selector === getComponentMode) {
+        return MODE_STATES.create;
+      }
+      return MODE_STATES.view;
+    });
+    const { getByText } = renderComponent();
+    expect(getByText(messages.cardCreate.defaultMessage)).toBeInTheDocument();
+  });
+
+  it('submits form with correct data in create mode', async () => {
+    const mockDispatch = jest.fn();
+    useDispatch.mockReturnValue(mockDispatch);
+
+    const { getByText, getByLabelText } = renderComponent();
+
+    await act(async () => {
```

Comment:
[important]: `userEvent` in its internal implementation [already has an act wrapper](https://github.com/testing-library/user-event/issues/497)